src/Valit/Exceptions/ValitExceptionExtensions.cs

```diff
@@ -1,20 +1,26 @@
+using System;
+
+using static Valit.Exceptions.SemanticExceptions;
+
 namespace Valit
 {
     public static class ValitExceptionExtensions
     {
-        internal static void ThrowIfNull(this object @object)
+        internal static void ThrowIfNull<T>(this T @object)
+            where T : class
```

Comment:
I think it is still possible to invoke that one on int (not nullable)

Reply:
Nope. Only on reference types - thanks, to that 'where' clause. It will not even work with Nullable<T>. You would need separate overload to handle these.
In the future, I suggest to use contracts instead as they can print simple expressions into log (such as 'rule != null') or could be guaranteed to be correct as part of static code analysis. Even better... Assume your code is null free and only check system inputs against nulls(CanBeNullAttribute). 
Inside the code use null alternatives as nullobjects, Result/Conditional/Optional objects, as these are easier to statically prove that they have been checked to not be in incorrect state. But it is a subject for some other time.

Here is some explanation for CanBeNull and static analysis: https://www.jetbrains.com/help/resharper/Code_Analysis__Value_Analysis.html